### DISCUSSION TITLE
Build failure for libsodium 1.0.17 on docker image debian: bullseye

### DISCUSSION THREAD

**Discussion by jsoneaday** [jedisct1/libsodium#1297] [Q&A]
I get this error below seems like it fails on pwhash_scrypt.

#0 252.8   FAIL: pwhash_scrypt
#0 252.8   ============================================================================
#0 252.8   Testsuite summary for libsodium 1.0.17
#0 252.8   ============================================================================
#0 252.8   # TOTAL: 75
#0 252.8   # PASS:  74
#0 252.8   # SKIP:  0
#0 252.8   # XFAIL: 0
#0 252.8   # FAIL:  1
#0 252.8   # XPASS: 0
#0 252.8   # ERROR: 0
#0 252.8   ============================================================================
#0 252.8   See test/default/test-suite.log
#0 252.8   Please report to https://github.com/jedisct1/libsodium/issues
#0 252.8   ============================================================================
#0 252.8   make[4]: Leaving directory '/usr/src/syntaxmakersserver/target/debug/build/exonum_libsodium-sys-0f0c3a9b3a75c0c1/out/source/libsodium-1.0.17/test/default'
#0 252.8   make[3]: Leaving directory '/usr/src/syntaxmakersserver/target/debug/build/exonum_libsodium-sys-0f0c3a9b3a75c0c1/out/source/libsodium-1.0.17/test/default'
#0 252.8   make[2]: Leaving directory '/usr/src/syntaxmakersserver/target/debug/build/exonum_libsodium-sys-0f0c3a9b3a75c0c1/out/source/libsodium-1.0.17/test/default'
#0 252.8   make[1]: Leaving directory '/usr/src/syntaxmakersserver/target/debug/build/exonum_libsodium-sys-0f0c3a9b3a75c0c1/out/source/libsodium-1.0.17/test'
#0 252.8 
#0 252.8   libtool: warning: '-version-info/-version-number' is ignored for convenience libraries
#0 252.8   ar: `u' modifier ignored since `D' is the default (see `U')
#0 252.8   libtool: warning: '-version-info/-version-number' is ignored for convenience libraries
#0 252.8   ar: `u' modifier ignored since `D' is the default (see `U')
#0 252.8   libtool: warning: '-version-info/-version-number' is ignored for convenience libraries
#0 252.8   ar: `u' modifier ignored since `D' is the default (see `U')
#0 252.8   libtool: warning: '-version-info/-version-number' is ignored for convenience libraries
#0 252.8   ar: `u' modifier ignored since `D' is the default (see `U')
#0 252.8   libtool: warning: '-version-info/-version-number' is ignored for convenience libraries
#0 252.8   ar: `u' modifier ignored since `D' is the default (see `U')
#0 252.8   libtool: warning: '-version-info/-version-number' is ignored for convenience libraries
#0 252.8   ar: `u' modifier ignored since `D' is the default (see `U')
#0 252.8   libtool: warning: '-version-info/-version-number' is ignored for convenience libraries
#0 252.8   ar: `u' modifier ignored since `D' is the default (see `U')
#0 252.8   ar: `u' modifier ignored since `D' is the default (see `U')
#0 252.8   ../../build-aux/test-driver: line 107: 55055 Killed                  "$@" > $log_file 2>&1
#0 252.8   make[4]: *** [Makefile:1893: test-suite.log] Error 1
#0 252.8   make[3]: *** [Makefile:2001: check-TESTS] Error 2
#0 252.8   make[2]: *** [Makefile:2593: check-am] Error 2
#0 252.8   make[1]: *** [Makefile:401: check-recursive] Error 1
#0 252.8   make: *** [Makefile:514: check-recursive] Error 1
#0 252.8 
#0 252.8   ', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/exonum_libsodium-sys-0.0.23/build.rs:264:13
#0 252.8   stack backtrace:
#0 252.8      0: rust_begin_unwind
#0 252.8                at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/std/src/panicking.rs:575:5
#0 252.8      1: core::panicking::panic_fmt
#0 252.8                at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/panicking.rs:64:14
#0 252.8      2: build_script_build::main
#0 252.8                at ./build.rs:264:13
#0 252.8      3: core::ops::function::FnOnce::call_once
#0 252.8                at /rustc/9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0/library/core/src/ops/function.rs:250:5

**Comment by jsoneaday** [comment]
I just tried switching to Ubuntu but same errors.

**Comment by jedisct1** [comment]
That issue was fixed years ago!

Looks like you are not compiling it directly, but some Rust library does.

Maybe try a more recent version of that Rust library, if it's not current?

Or install the Debian (or Ubuntu) package for `libsodium`, and maybe that library is going to just use it instead of trying to recompile libsodium.

There's also a Rust crate called [`libsodium-sys-stable`](https://crates.io/crates/libsodium-sys-stable) that seems to do the same thing, but is updated more frequently.

**Reply by jsoneaday** [reply]
Yeah Im using another crate exonum-crypto so just letting it run whatever it needs. Not sure what to do on this
